paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/commands/WriteIntoPaimonTable.scala

```diff
@@ -113,7 +120,9 @@ case class WriteIntoPaimonTable(table: FileStoreTable, overwrite: Boolean, data:
         .map(deserializeCommitMessage(serializer, _))
 
     try {
-      val tableCommit = writeBuilder.newCommit().asInstanceOf[InnerTableCommit]
+      val tableCommit = writeBuilder
+        .withOverwrite(overwritePartition.asJava)
```

Comment:
if there is no overwrite, we should not invoke `withOverwrite(emptyMap)` here?
I don't understand why tests passing.

Reply:
if no overwrite here, `overwritePartition` is actually `null`, that is consistent with the behavior of `insert into`.

Reply:
I get it, you are right.